prov/tcp/src/tcpx_eq.c

```diff
@@ -42,13 +42,9 @@ static ssize_t tcpx_eq_read(struct fid_eq *eq_fid, uint32_t *event,
 
 	eq = container_of(eq_fid, struct util_eq, eq_fid);
 
-	fastlock_acquire(&eq->lock);
-	if (slist_empty(&eq->list)) {
-		fastlock_release(&eq->lock);
+	if (slist_empty(&eq->list))
```

Comment:
I don't think we need this check at all.  Reading the EQ isn't supposed to be a fast path operation, so always calling tcpx_conn_mgr_run() should be okay.

Reply:
@shefty Done.